reactor-core/src/main/java/reactor/Flux.java

```diff
@@ -0,0 +1,250 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package reactor;
+
+import java.util.concurrent.BlockingQueue;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import reactor.core.publisher.PublisherFactory;
+import reactor.core.support.Assert;
+import reactor.fn.BiConsumer;
+import reactor.fn.BiFunction;
+import reactor.fn.Consumer;
+import reactor.fn.Function;
+
+/**
+ * A Reactive Streams {@link Publisher} with basic rx operators that emits 0 to N elements,
+ * and then complete (successfully or with an error).
+ *
+ * <p>It is intended to be used in Reactive Spring projects implementation and return types.
+ * Input parameters should keep using raw {@link Publisher} as much as possible.
+ *
+ * <p>If it is known that the underlying {@link Publisher} will emit 0 or 1 element,
+ * {@link Mono} should be used instead.
+ *
+ * @author Sebastien Deleuze
+ * @since 2.5
+ * @see Mono
+ */
+public class Flux<T> implements Publisher<T> {
+
+	private final Publisher<T> publisher;
+
+
+	protected Flux(Publisher<T> publisher) {
+		Assert.notNull(publisher);
+		this.publisher = publisher;
+	}
+
+	/**
+	 * Create a new {@code Flux} that emits the specified item.
+	 */
+	public static <T> Flux<T> just(T data) {
+		return new Flux<>(Publishers.just(data));
+	}
+
+	/**
+	 * Expose the specified {@link Publisher} with the {@code Flux} API.
+	 */
+	public static <T> Flux<T> from(Publisher<T> source) {
+		return (source instanceof Flux ? (Flux<T>)source :  new Flux<>(source));
+	}
+
+	/**
+	 * Create a {@code Flux} that emits the items contained in the provided {@link Iterable}.
+	 */
+	public static <T> Flux<T> from(Iterable<? extends T> it) {
+		return new Flux<>(Publishers.from(it));
+	}
+
+	/**
+	 * Create a {@code Flux} reacting on each available {@link Subscriber} read derived with the
+	 * passed {@link Consumer}. If a previous request is still running, avoid recursion and extend
+	 * the previous request iterations.
+	 */
+	public static <T> Flux<T> from(Consumer<Subscriber<? super T>> requestConsumer) {
+		return new Flux<>(PublisherFactory.create(c -> requestConsumer.accept(c)));
+	}
+
+	/**
+	 * Create a {@code Flux} that completes with the specified error.
+	 */
+	public static <T> Flux<T> error(Throwable error) {
+		return new Flux<>(Publishers.error(error));
+	}
+
+	/**
+	 * Create a {@code Flux} that completes without emitting any item.
+	 */
+	public static <T> Flux<T> empty() {
+		return new Flux<>(Publishers.empty());
```

Comment:
You don't need to create an empty Flux all the time (I assume Flux is otherwise stateless), create a static final Flux<Object> EMPTY once and cast+return it from this method.